lib/generators/has_friendship/has_friendship_generator.rb

```diff
@@ -9,7 +9,10 @@ def self.source_root
   end
 
   def self.next_migration_number(path)
-    Time.now.utc.strftime("%Y%m%d%H%M%S")
+    next_num = Time.now.utc.strftime("%Y%m%d%H%M%S").to_i
+    current_num = current_migration_number(path)
+    next_num += (current_num - next_num + 1) if current_num >= next_num
```

Comment:
Here’s the deal.  This method is so fast, that even 10 calls to migration_template could all happen within the same second.  So next_num will stay the same (since it’s the datetime stamp) but current_num keeps incrementing with every call.  Therefore next_num shouldn’t always be current_num + 1. You have to get the difference between the two, add one to that, and then INCREMENT next_num by that value.

Once you wrap your head around how it all works, it’s pretty simple and works well.  Like I said before, you could add a sleep(1) between calls, or inside the call to next_migration_number but it seems lame to deliberately make your gem install go slower. You could also add the milliseconds on to the end of the version number, but then the version number format isn’t consistent with how most other migration version numbers are formatted.